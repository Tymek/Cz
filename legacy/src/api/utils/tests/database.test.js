import fs from 'fs'
import { identity } from 'ramda'

import createDatabase, { reset } from '../database'

const restoreFS = (() => {
  const {
    existsSync,
    writeFileSync,
  } = fs

  return () => {
    fs.existsSync = existsSync
    fs.writeFileSync = writeFileSync
  }
})()

afterEach(reset)

describe('server/utils/db', () => {
  it('create new database file', () => {
    let output

    fs.existsSync = jest.fn().mockReturnValue(false)
    fs.writeFileSync = jest.fn()
      .mockImplementation((file, input) => { output = input })

    const db = createDatabase('test')
    db.set('value')

    restoreFS()
    return expect(output).toMatchSnapshot()
  })

  it('reads and writes to and from database', () => {
    fs.writeFileSync = jest.fn().mockImplementation(identity)

    const db = createDatabase('test')
    db.set('value')

    restoreFS()
    return expect(db.get()).toEqual('value')
  })

  it('reads and writes to and from specified path in database', () => {
    fs.writeFileSync = jest.fn().mockImplementation(identity)

    const db = createDatabase('test')
    const path = ['subkey']
    db.set('value', path)

    restoreFS()
    return expect(db.get(path)).toEqual('value')
  })

  it('saves complex path', () => {
    let output
    fs.writeFileSync = jest.fn()
      .mockImplementation((file, input) => { output = input })

    const db = createDatabase('different_test')
    const path = ['path', 'to', 'set']
    db.set({
      value: [],
    }, path)

    restoreFS()
    return expect(output).toMatchSnapshot()
  })
})
