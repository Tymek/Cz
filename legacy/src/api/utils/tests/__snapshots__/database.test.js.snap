// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server/utils/db create new database file 1`] = `
"{
  \\"test\\": \\"value\\"
}"
`;

exports[`server/utils/db saves complex path 1`] = `
"{
  \\"different_test\\": {
    \\"path\\": {
      \\"to\\": {
        \\"set\\": {
          \\"value\\": []
        }
      }
    }
  }
}"
`;
